op-batcher/flags/throttle_flags.go

```diff
@@ -0,0 +1,171 @@
+	DefaultPIDSampleTime          = 2 * time.Second
+	DefaultPIDKp                  = 0.33
+	DefaultPIDKi                  = 0.01
+	DefaultPIDKd                  = 0.05
+	DefaultPIDIntegralMax         = 1000.0
+	DefaultPIDOutputMax           = 1.0
+)
+
+var (
+	// Block-builder side
+	AdditionalThrottlingEndpointsFlag = &cli.StringSliceFlag{
+		Name:    "throttle.additional-endpoints",
+		Usage:   "Comma-separated list of endpoints to distribute throttling configuration to (in addition to the L2 endpoints specified with --l2-eth-rpc).",
+		EnvVars: prefixEnvVars("THROTTLE_ADDITIONAL_ENDPOINTS"),
+	}
+
+	// Builder-side Tx-size limits
+	ThrottleTxSizeLowerLimitFlag = &cli.IntFlag{
+		Name:    "throttle.tx-size-lower-limit",
+		Usage:   "The limit on the DA size of transactions when we are at maximum throttle intensity",
+		Value:   DefaultThrottleTxSizeLowerLimit,
+		EnvVars: prefixEnvVars("THROTTLE_TX_SIZE_LOWER_LIMIT"),
+	}
+	ThrottleTxSizeUpperLimitFlag = &cli.IntFlag{
+		Name:    "throttle.tx-size-upper-limit",
+		Usage:   "The limit on the DA size of transactions when we are at 0+ throttle intensity (limit of the intensity as it approaches 0 from positive values). Not applied when throttling is inactive.",
+		Value:   DefaultThrottleTxSizeUpperLimit,
+		EnvVars: prefixEnvVars("THROTTLE_TX_SIZE_UPPER_LIMIT"),
+	}
+
+	// Builder-side block-size limits
+	ThrottleBlockSizeLowerLimitFlag = &cli.IntFlag{
+		Name:    "throttle.block-size-lower-limit",
+		Usage:   "The limit on the DA size of blocks when we are at maximum throttle intensity (linear and quadratic controllers only)",
+		Value:   DefaultThrottleBlockSizeLowerLimit,
+		EnvVars: prefixEnvVars("THROTTLE_BLOCK_SIZE_LOWER_LIMIT"),
+	}
+	ThrottleBlockSizeUpperLimitFlag = &cli.IntFlag{
+		Name:    "throttle.block-size-upper-limit",
+		Usage:   "The limit on the DA size of blocks when we are at 0 throttle intensity (applied when throttling is inactive)",
+		Value:   DefaultThrottleBlockSizeUpperLimit,
+		EnvVars: prefixEnvVars("THROTTLE_BLOCK_SIZE_UPPER_LIMIT"),
+	}
+
+	// // Controller side
+	ThrottleControllerTypeFlag = &cli.StringFlag{
+		Name:    "throttle.controller-type",
+		Usage:   "Type of throttle controller to use: 'step', 'linear', 'quadratic' (default) or 'pid' (EXPERIMENTAL - use with caution)",
+		Value:   DefaultThrottleControllerType,
+		EnvVars: prefixEnvVars("THROTTLE_CONTROLLER_TYPE"),
+		Action: func(ctx *cli.Context, value string) error {
+			validTypes := []string{"step", "linear", "quadratic", "pid"}
+			for _, validType := range validTypes {
+				if value == validType {
+					return nil
+				}
+			}
+			return fmt.Errorf("throttle-controller-type must be one of %v, got %s", validTypes, value)
+		},
+	}
+	ThrottleUsafeDABytesLowerThresholdFlag = &cli.IntFlag{
+		Name:    "throttle.unsafe-da-bytes-lower-threshold",
+		Usage:   "The threshold on unsafe_da_bytes beyond which the batcher will start to throttle the block builder. Zero disables throttling.",
```

Comment:
There's an opportunity here to clean up the way we disable throttling. IMO cleaner to have a "disabled" controller or other more obvious way to disable rather than a magic value here.

Reply:
Taking that on here https://github.com/ethereum-optimism/optimism/pull/17067